impeller/renderer/backend/metal/surface_mtl.mm

```diff
@@ -224,6 +225,7 @@
   if (!context) {
     return false;
   }
+  context->DidFinishSurfaceFrame();
```

Comment:
Do we need to call these in the playground and between golden PNG generation in order to not leak/risk GPU OOM?